core/src/main/scala/chisel3/experimental/dataview/DataView.scala

```diff
@@ -71,13 +71,11 @@ sealed class DataView[T: DataProduct, V <: Data] private[chisel3] (
     * @return a new `DataView` from the original `T` to new view-type `V2`
     */
   def andThen[V2 <: Data](g: DataView[V, V2])(implicit sourceInfo: SourceInfo): DataView[T, V2] = {
-    val self = this
-    // We have to pass the DataProducts and DataViews manually to .viewAs below
-    val tdp = implicitly[DataProduct[T]]
-    val vdp = implicitly[DataProduct[V]]
+    implicit val self: DataView[T, V] = this
+    implicit val gg:   DataView[V, V2] = g
     new DataView[T, V2](
       t => g.mkView(mkView(t)),
-      { case (t, v2) => List(t.viewAs[V](tdp, self).viewAs[V2](vdp, g) -> v2) },
+      { case (t, v2) => List(t.viewAs[V].viewAs[V2] -> v2) },
```

Comment:
When I added source info as an implicit argument to `.viewAs` it told me I need to pass it explicitly here (which is true) but I realized that it would be simpler just to mark the two DataViews as implicit. When I originally wrote this I think I thought there would be ambiguity issues with the implicit DataViews but there obviously isn't because they are of different types. I added the type annotations just to make it more clear.